Tandoor Won't Update

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

Tandoor

### 📂 What was the exact command used to execute the script?

update

### 📝 Provide a clear and concise description of the issue.

See code below.  Server won't properly update

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Ubuntu 22.04

### 🔄 Steps to reproduce the issue.

perform update

### ❌ Paste the full error output (if available).

⠹ Updating Tandoor (Patienceerror
  error: subprocess-exited-with-error
  
  × Building wheel for xmlsec (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [30 lines of output]
      WARNING setuptools_scm.pyproject_reading toml section missing 'pyproject.toml does not contain a tool.setuptools_scm section'
      Traceback (most recent call last):
        File "/tmp/pip-build-env-vznt0o0h/overlay/local/lib/python3.11/dist-packages/setuptools_scm/_integration/pyproject_reading.py", line 36, in read_pyproject
          section = defn.get("tool", {})[tool_name]
                    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
      KeyError: 'setuptools_scm'
      /tmp/pip-build-env-vznt0o0h/overlay/local/lib/python3.11/dist-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!
      
              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:
      
              License :: OSI Approved :: MIT License
      
              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************
      
      !!
        self._finalize_license_expression()
      running bdist_wheel
      running build
      running build_py
      creating build/lib.linux-x86_64-cpython-311/xmlsec
      copying src/xmlsec/template.pyi -> build/lib.linux-x86_64-cpython-311/xmlsec
      copying src/xmlsec/__init__.pyi -> build/lib.linux-x86_64-cpython-311/xmlsec
      copying src/xmlsec/constants.pyi -> build/lib.linux-x86_64-cpython-311/xmlsec
      copying src/xmlsec/tree.pyi -> build/lib.linux-x86_64-cpython-311/xmlsec
      copying src/xmlsec/py.typed -> build/lib.linux-x86_64-cpython-311/xmlsec
      running build_ext
      error: Unable to invoke pkg-config.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for xmlsec
Successfully built django-allauth
Failed to build xmlsec
ERROR: Could not build wheels for xmlsec, which is required to install pyproject.toml-based projects
⠦ Updating Tandoor (Patience)
[ERROR] in line 36: exit code 0: while executing command $STD pip3 install -r requirements.txt

### 🖼️ Additional context (optional).

_No response_

--- a/ct/tandoor.sh
+++ b/ct/tandoor.sh
@@ -27,20 +27,24 @@ function update_script() {
     msg_error "No ${APP} Installation Found!"
     exit
   fi
+  if ! [[ $(dpkg -s python3-xmlsec 2>/dev/null) ]]; then
+    $STD apt-get update
+    $STD apt-get install -y python3-xmlsec
+  fi
   if cd /opt/tandoor && git pull | grep -q 'Already up to date'; then
     msg_ok "There is currently no update available."
   else
     msg_info "Updating ${APP} (Patience)"
     export $(cat /opt/tandoor/.env | grep "^[^#]" | xargs)
-    cd /opt/tandoor/
+    cd /opt/tandoor/ || exit
     $STD pip3 install -r requirements.txt
     $STD /usr/bin/python3 /opt/tandoor/manage.py migrate
     $STD /usr/bin/python3 /opt/tandoor/manage.py collectstatic --no-input
     $STD /usr/bin/python3 /opt/tandoor/manage.py collectstatic_js_reverse
-    cd /opt/tandoor/vue
+    cd /opt/tandoor/vue || exit
     $STD yarn install
     $STD yarn build
-    cd /opt/tandoor
+    cd /opt/tandoor || exit
     $STD python3 version.py
     systemctl restart gunicorn_tandoor
     msg_ok "Updated ${APP}"

--- a/install/tandoor-install.sh
+++ b/install/tandoor-install.sh
@@ -38,7 +38,8 @@ $STD apt-get install -y \
   python3 \
   python3-dev \
   python3-setuptools \
-  python3-pip
+  python3-pip \
+  python3-xmlsec
 rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED
 msg_ok "Setup Python3"
 

